src/libsyntax/parse/parser.rs

```diff
@@ -2700,6 +2700,17 @@ impl<'a> Parser<'a> {
                 let (span, e) = self.interpolated_or_expr_span(e)?;
                 (span, self.mk_unary(UnOp::Not, e))
             }
+            // fix issue #41679: Suggest `!` for bitwise negation when encountering a `~`
```

Comment:
Nit: could you remove "fix issue #41679:"? (Not useful for a reader and the history is available from git if necessary.)